pandas/tools/tests/test_tile.py

```diff
@@ -313,6 +313,18 @@ def test_datetime_cut(self):
         result, bins = cut(data, 3, retbins=True)
         tm.assert_series_equal(Series(result), expected)
 
+    def test_datetime_bin(self):
+        data = [np.datetime64('2012-12-13'), np.datetime64('2012-12-15')]
+        bins = [np.datetime64('2012-12-12'), np.datetime64('2012-12-14'),
```

Comment:
actually you prob want to test with ``datetime.datetime``, ``Timestamp``, ``np.datetime64`` (but just put them in a loop something like

data = ['2012-12-12', '2012-12-14']

for conv in [Timestamp(x).to_pydatetime, Timestamp, np.datetime64]:
      bins = [ conv(v) for v in data ]
```
also test
``bins = pd.to_datetime(data)``

these should all work, because internally you need to wrap a ``Timestamp`` converter around each of fhe bins (if dtype==M8) or ``Timedelta`` if dtype==m8